sqlmesh/core/snapshot/evaluator.py

```diff
@@ -1661,6 +1663,83 @@ def get_custom_materialization_type(name: str) -> t.Type[CustomMaterialization]:
     return strategy_type
 
 
+class EngineManagedStrategy(PromotableStrategy):
+    def create(
+        self,
+        table_name: str,
+        model: Model,
+        is_table_deployable: bool,
+        **render_kwargs: t.Any,
+    ) -> None:
+        rendered_query = model.render_query_or_raise(**render_kwargs)
+
+        # only create the managed table if this snapshot can actually be deployed to prod (eg if an upstream model doesnt have a forward-only change)
+        # vendors providing managed tables (such as Snowflake's dynamic tables) tend to charge more for them, they arent as cheap as normal tables
+        if is_table_deployable:
```

Comment:
I'm still not 100% clear on what to do here.

I notice that the SnapshotEvaluator calls `create()` twice, once for the `_temp` version (`is_table_deployable=False`) and then immediately afterwards for the actual version `is_table_deployable=True`.

The assumption is that the `_temp` version can just be a normal table because it would only ever be populated with preview data. However, I still don't fully understand how this all ties in with the case where the production table gets shallow cloned in order to test a forward-only change in a non-prod environment